### DISCUSSION TITLE
Can multiple queries share the same key?

### DISCUSSION THREAD

**Discussion by geekox86** [TanStack/query#3776] [Q&A] [answered]
Greetings all,

What happens when multiple queries share the same key in the following cases:

1- When I call queryClient.refetchQueries with this key? Which query will fire?

2- When they return different results with different structures?

3- When they return the same result structure but only partially with some missing fields? Is there a custom result merger option?

Many thx in advance!

**Comment by TkDodo** [comment, marked as answer]
queryKeys and queryFn come in pairs. The cache is just a js object where the key of the object is the serialized queryKey, and the value of that entry in the object is the awaited result of the `queryFn`.

So you can't really have the same key with different query functions that produce different results. If you produce a different result, please use a separate key, as keys have to be unique.
